dpu-api/device_service.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package Vendor;
+
+option go_package = "github.com/openshift/dpu-operator/api/dpu-api";
+
+import "google/protobuf/empty.proto";
+
+// The device plugin API service definition.
+service DeviceService {
+  // GetDevices returns a list of devices.
+  rpc GetDevices(google.protobuf.Empty) returns (DeviceListResponse);
+}
+
+message TopologyInfo {
+  string node = 1;
+}
+
+// Device represents a single device.
+message Device {
+  string ID = 1;
+  string health = 2;
+  TopologyInfo topology = 3;
+}
+
+// DeviceListResponse is the response message containing the devices map.
+message DeviceListResponse {
+  map<string, Device> devices = 1;
```

Comment:
Can you explain why we need map instead of `repeated`

Reply:
good question - this is just how I logically did it
we did not define a proto definition, so this is what I went with.

would you prefer if it was ``repeated`` ?

Reply:
It's not that I prefer `map` or `repeated`. Rather, I want to check if there is a real reason for map? Do we have keys + values? Or is it "just a list of values".

Reply:
it's key + values, as the device could have more info than just the ID.